lib/cartopy/mpl/geoaxes.py

```diff
@@ -1964,14 +1956,8 @@ def add_wms(self, wms, layers, wms_kwargs=None, **kwargs):
 
 # Define the GeoAxesSubplot class, so that a type(ax) will emanate from
 # cartopy.mpl.geoaxes, not matplotlib.axes.
-class GeoAxesSubplot(matplotlib.axes.SubplotBase, GeoAxes):
-    _axes_class = GeoAxes
-
-
-try:
-    matplotlib.axes._subplots._subplot_classes[GeoAxes] = GeoAxesSubplot
-except AttributeError:
-    matplotlib.axes._subplot_classes[GeoAxes] = GeoAxesSubplot
+GeoAxesSubplot = matplotlib.axes.subplot_class_factory(GeoAxes)
+GeoAxesSubplot.__module__ = GeoAxes.__module__
```

Comment:
Did something change?

Reply:
See the comment above; this ensures that we get `cartopy.mpl.geoaxes.GeoAxesSubplot` instead of some matplotlib path (since by using the `subplot_class_factory`, it's created there.)

Reply:
I *wrote* that comment, but didn't write the ``__module__`` part 😃 - so was getting some clarification.

Reply:
Ah, well, not using the private API necessitated a small change in how the module path was fixed.